nncf/common/utils/patcher.py

```diff
@@ -0,0 +1,157 @@
+"""
```

Comment:
> Note: The patcher is not general though, for example it doesn't support patching with a function defined in another function's scope, patching partial functions, repatching static or class methods twice, repatching regular methods thrice.
> Covered the Patcher by tests

[patcher](https://github.com/openvinotoolkit/training_extensions/blob/develop/otx/core/patcher.py)
[unit test for patcher](https://github.com/openvinotoolkit/training_extensions/blob/develop/tests/unit/core/test_core_patcher.py)
`Patcher` is now can repatch methods, static methods, class methods, functions, and even patch them recursively.

Reply:
Thanks for the feedback. I've updated the Patcher class with your changes.